### DISCUSSION TITLE
Hydration on synchronous storage isn't really synchronous

### DISCUSSION THREAD

**Discussion by AShapi** [pmndrs/zustand#1609] [General] [answered]
when using `createStore` from `zustand/vanilla` with a synchronous storage (localStorage).
I would expect the store to be hydrated synchronously

```ts
// state.js
export const someStore = createStore(persist((set, get) => ({
... 

// file.js
import { someStore } from 'state'
console.log(regionStore.getState().getSome()) // should be hydrated already but isn't
```

I think the issue lays behind using promises for the hydration even for synchronous storages
and thus creating a microtask delay for the hydration.

![image](https://user-images.githubusercontent.com/84472699/217596835-67226d1f-87b8-42d7-bcab-a1a1ea4b451c.png)

**Comment by dai-shi** [comment, marked as answer]
I thought #1518 fixed it, but do you still see issues? (I remember someone says there's potential issues in the current implementation, so it's not too surprising.)

Can you create a test case and fix it?

**Reply by AShapi** [reply]
It did, i wasn't updated.
Thanks!
